api/v1/firewall_types.go

```diff
@@ -78,6 +78,8 @@ type Data struct {
 	EgressRules []EgressRuleSNAT `json:"egressRules,omitempty"`
 	// FirewallNetworks holds the networks known at the metal-api for this firewall machine
 	FirewallNetworks []FirewallNetwork `json:"firewallNetworks,omitempty"`
+	// DNSServerAddress specifies DNS server address used by DNS proxy
+	DNSServerAddress string `json:"dnsServerAddress,omitempty"`
```

Comment:
It should be validated on startup / change whether the address is actually reachable.